src/main/java/com/intel/genomicsdb/GenomicsDBImporter.java

```diff
@@ -23,1615 +23,1665 @@
 import org.json.simple.JSONObject;
 import org.json.simple.parser.JSONParser;
 import org.json.simple.parser.ParseException;
-import java.io.IOException;
-import java.io.FileNotFoundException;
-import java.io.File;
-import java.io.PrintWriter;
-import java.io.StringWriter;
+
+import java.io.*;
 import java.util.*;
+import java.util.stream.Collectors;
 
 import static com.googlecode.protobuf.format.JsonFormat.printToString;
 
 /**
  * Java wrapper for vcf2tiledb - imports VCFs into TileDB/GenomicsDB.
  * All vid information is assumed to be set correctly by the user (JSON files)
  */
-public class GenomicsDBImporter
-{
-  static
-  {
-    try
-    {
-      boolean loaded = GenomicsDBUtils.loadLibrary();
-      if(!loaded)
-        throw new GenomicsDBException("Could not load genomicsdb native library");
-    }
-    catch(UnsatisfiedLinkError ule)
-    {
-      throw new GenomicsDBException("Could not load genomicsdb native library", ule);
+public class GenomicsDBImporter {
+    private static final String DEFAULT_ARRAY_NAME = "genomicsdb_array";
+    private static final int DEFAULT_TILEDB_CELLS_PER_TILE = 1000;
+    //Allele specific annotation fields
+    private static final HashSet<String> mRLengthHistogramFieldsWithFloatBins = new HashSet<>(Arrays.asList(
+            "AS_RAW_BaseQRankSum",
+            "AS_RAW_MQRankSum",
+            "AS_RAW_ReadPosRankSum"
+    ));
+    private static final HashSet<String> mRLengthTwoDFloatVectorFields = new HashSet<>(Collections.singletonList(
+            "AS_RAW_MQ"
+    ));
+    private static final HashSet<String> mRLengthTwoDIntVectorFields = new HashSet<>(Collections.singletonList(
+            "AS_SB_TABLE"
+    ));
+    static final String CHROMOSOME_INTERVAL_FOLDER = "%s_%d_%d";
+    static long mDefaultBufferCapacity = 20480; //20KB
+
+    static {
+        try {
+            boolean loaded = GenomicsDBUtils.loadLibrary();
+            if (!loaded) throw new GenomicsDBException("Could not load genomicsdb native library");
+        } catch (UnsatisfiedLinkError ule) {
+            throw new GenomicsDBException("Could not load genomicsdb native library", ule);
+        }
     }
-  }
-
-  static long mDefaultBufferCapacity = 20480; //20KB
-  private static final String DEFAULT_ARRAYNAME = "genomicsdb_array";
-  private static final int DEFAULT_TILEDB_CELLS_PER_TILE = 1000;
-
-  //Allele specific annotation fields
-  public static final HashSet<String> mRLengthHistogramFieldsWithFloatBins = new HashSet<>(Arrays.asList(
-              "AS_RAW_BaseQRankSum",
-              "AS_RAW_MQRankSum",
-              "AS_RAW_ReadPosRankSum"
-              ));
-
-  public static final HashSet<String> mRLengthTwoDFloatVectorFields = new HashSet<>(Arrays.asList(
-              "AS_RAW_MQ"
-              ));
-
-  public static final HashSet<String> mRLengthTwoDIntVectorFields = new HashSet<>(Arrays.asList(
-              "AS_SB_TABLE"
-              ));
 
   /*
    * JNI functions
    */
```

Comment:
What's the point of moving all the JNI functions down?

Reply:
This was done using the IDE which makes use of best standard practices in Java. In this case, ctors should be the first functions to appear in the file. Next time, I'm going to split the work on the format in order to avoid confusion.

Reply:
I'm created a new user story (https://jira01.devtools.intel.com/browse/AEJ-271) to reduce the amount of code in this class.